src/main/java/com/epam/webapphello/service/UserServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.webapphello.service;
+
+import com.epam.webapphello.dao.DaoHelper;
+import com.epam.webapphello.dao.DaoHelperFactory;
+import com.epam.webapphello.dao.UserDao;
+import com.epam.webapphello.entity.User;
+import com.epam.webapphello.exception.DAOException;
+import com.epam.webapphello.exception.ServiceException;
+
+
+import java.util.Optional;
+
+public class UserServiceImpl implements UserService {
+
+    private DaoHelperFactory daoHelperFactory;
+
+    public UserServiceImpl(DaoHelperFactory daoHelperFactory) {
+        this.daoHelperFactory = daoHelperFactory;
+    }
+
+    @Override
+    public Optional<User> login(String login, String password) throws ServiceException {
+        Optional<User> user = null;
+        try (DaoHelper helper = daoHelperFactory.create()) {
+            helper.startTransaction();
```

Comment:
no need in transaction here since you update nothing and query only one table

Reply:
i understand

Reply:
resolved comments that you addressed

Reply:
ok